virttest/utils_env.py

```diff
@@ -314,69 +335,92 @@ def get_lvmdev(self, name):
         """
         return self.data.get("lvmdev__%s" % name)
 
-    def _start_tcpdump(self):
-
-        cmd_template = "%s -npvvvi any 'port 68 or port 546'"
-        if self._params.get("remote_preprocess") == "yes":
+    def _start_sniffer(self):
+        ip_s = [("tshark",
+                 "%s -npVi any 'port 68 or port 546' -T fields -E header=y "
+                 "-E separator=/s -E occurrence=f -e ip.src -e ip.dst "
+                 "-e bootp.type -e bootp.id -e bootp.ip.client "
+                 "-e bootp.hw.mac_addr -e bootp.ip.your -e bootp.option.dhcp",
+                 _output_handler_tshark),
+                ("tcpdump",
+                 "%s -npvvvi any 'port 68 or port 546'",
+                 _output_handler_tcpdump)]
```

Comment:
This is actually the only thing that troubles me. When there was just a single sniffer, it was acceptable to have it hardcoded, but now with multiple choices it'd be way cleaner to implement them as objects. Would it be too much to ask? It'd improve the code organization significantly.

Reply:
Also I'm not sure how to deal with the IPv6. Some tests require that and in such case the `tshark` should not be used. How about running both? If I understand the problem correctly `tcpdump` might not update on some occasions, but it should not provide bad results, right? Then running both should not break anything.

Reply:
As for the object rework, I'd imagine something like:

```
class Sniffer(object):
    def __init__(self, params)  # store session information and so on
    def start()   # common code to start sniffer (remotely/locally based on params) and report success/failure
    def stop()    # stop sniffer
    def is_alive()

class TcpdumpSniffer(Sniffer):

class TSharkSniffer(Sniffer):

...

def _start_sniffer(self):
    for sniffer_cls in (TcpdumpSniffer, TSharkSniffer):
        sniffer = sniffer_cls(params)
        if sniffer.start():
            break
...
```

Reply:
OK, make sense, have implemented this in v2.